diagrams/request_uri_mode_create.md

```diff
@@ -0,0 +1,49 @@
+```plantuml
+@startuml
+
+autonumber
+
+box "Wallet"
+participant "Metadata" as wm
+participant "Authorization Endpoint" as w
+end box
+
+participant "User Agent" as u
+
+box "Verifier"
+participant "Frontend" as r
+participant "Request Endpoint" as rp
+participant "Response Endpoint" as rb
+end box
+
+u --> r : use
+activate r
+
+r --> u: **signed authorization request**\n(client_id, request_uri, request_uri_mode=create, state)
```

Comment:
1. Not clear why request in this step has to be signed. This does not add any security. I'd rather provide the Authz Request parameters unsigned. NOTE: JAR does not sign `client_id` either if `request_uri` (or `request`) is used in the Authorization Request.
2. I believe there is a typo in this step: `request_uri_mode=create`, it should rather be `request_uri_mode=POST`.
3. I would rename the `request_uri_mode=POST` to `request_uri_mode=post_metadata_exchange`. This indicates additional AS metadata is provided plus a `w_nonce`.

Reply:
few remarks

1. having the request_uri where the singed request can be obtained, this http response (not a request) should be unsigned

2. `request_uri_mode` -> `request_uri_methods`, type Array, enabling one or more methods, eg: GET and POST. I'd move this information in the RP's metadata if we don't have any particular reason to keep it here in the request.

Reply:
@awoie I got perfectly your point, however I would not mix http methods with applicative of functional scopes like the metadata exchange.

my proposal is simple @tlodderstedt 

````
r --> u: **response**\n(client_id, request_uri, discovery_uri, state)
````

if the RP supports the wallet metadata discovery, it provides the discovery_uri (method POST)
if the wallet supports the discovery, it provides its metadata to the RP using http POST application/json

if at least no one supports the wallet metadata discovery the only choice is going ahead to the request_uri endpoint with a presumed common set of features.

nothing else, I hope that you find it easy as me as when I was thinking to it.

Reply:
> 1. Not clear why request in this step has to be signed. This does not add any security. I'd rather provide the Authz Request parameters unsigned. NOTE: JAR does not sign specifically the `client_id` either if `request_uri` is used in the Authorization Request. The `client_id` has to be always present in the query string parameters.

The signed request allows the wallet to authenticate the verifier before performing any further processing of the request. So the wallet may decide not to send a request to the request uri if it cannot trust the sender.

Reply:
> 3\. I would rename the `request_uri_mode=POST` to `request_uri_mode=post_metadata_exchange` or `pmx` (as in post metadata exchange request URI mode). This indicates additional AS metadata is provided plus a `w_nonce`.

How would you designate the "GET" method?

Reply:
> @awoie I got perfectly your point, however I would not mix http methods with applicative or functional scopes like the metadata exchange.
> 
> my proposal is simple @tlodderstedt
> 
> ```
> r --> u: **response**\n(client_id, request_uri, discovery_uri, state)
> ```
> 
> if the RP supports the wallet metadata discovery, it provides the discovery_uri (method POST) if the wallet supports the discovery, it provides its metadata to the RP using http POST application/json
> 
> if at least no one supports the wallet metadata discovery the only choice is going ahead to the request_uri endpoint with a presumed common set of features.
> 
> nothing else, I hope that you find it easy as me as when I was thinking to it.

no need for sending two uris. if the wallet does not support the new method, it will ignore the request_uri_mode parameter and just send a GET to the request uri.

Reply:
> > 1. Not clear why request in this step has to be signed. This does not add any security. I'd rather provide the Authz Request parameters unsigned. NOTE: JAR does not sign specifically the `client_id` either if `request_uri` is used in the Authorization Request. The `client_id` has to be always present in the query string parameters.
> 
> The signed request allows the wallet to authenticate the verifier before performing any further processing of the request. So the wallet may decide not to send a request to the request uri if it cannot trust the sender.

This is only true if the wallet has some means of getting a trustworthy public key of the sender i.e. a trust infrastructure is present. So in general it is not that useful. OTOH, the signature is not necessary with a trust infrastructure because the wallet can assume the sender is telling the truth about its identity, and get the sender's metadata from the identity endpoint provided by the trust infrastructure. i.e. the wallet does not need to believe anything that an unknown sender tells it. And having a signed message from an unknown sender whose public key you cannot validate because the wallet does not have a trust infrastructure, provides no security except message integrity.

Reply:
> no need for sending two uris. if the wallet does not support the new method, it will ignore the request_uri_mode parameter and just send a GET to the request uri.

@tlodderstedt ok, I approve this and I'm going to implement this. We'll continue with the definition of the details with this PR